gk/main.c

```diff
@@ -745,6 +822,22 @@ gk_stage2(void *arg)
 	if (ret < 0)
 		goto cleanup;
 
+	if (ipv6_if_configured(&gk_conf->net->front)) {
+		int i;
+		for (i = 0; i < gk_conf->num_lcores; i++) {
+			unsigned int lcore = gk_conf->lcores[i];
+			ret = register_ipv6_acl(&gk_conf->net->front,
+				lcore, GATEKEEPER_IPV6_ACL_ND);
+			if (ret < 0) {
+				RTE_LOG(ERR, GATEKEEPER,
+					"gk: could not register ND IPv6 ACL for lcore %u\n",
+					lcore);
+				goto cleanup;
+			}
+		}
+		build_ipv6_acls(&gk_conf->net->front);
```

Comment:
I tried doing it that way, but there is a tricky sequence that needs to be observed. Calculating all the IPv6 addresses requires knowing the Ethernet address of the NIC, which can only be retrieved after the NIC is started (in stage 2). So the blocks that need the ACL are forced to register in stage 2. It would be nice to be able to build the ACLs just once in stage 3, but stage 3 is for launching the lcores of the blocks -- in other words, there's no stage 3 for the network configuration. So I think we could either change stage 3 to allow the ACL to be built once, or just allow the ACL to be re-built if needed when new registrations happen in stage 2.